tests/currency_network/test_currency_network_interests.py

```diff
@@ -805,3 +855,94 @@ def test_apply_interests_not_possible_when_frozen(
     adapter.update_trustline(A, B, is_frozen=True, accept=True)
     with pytest.raises(TransactionFailed):
         adapter.contract.functions.applyInterests(B).transact({"from": A})
+
+
+# Above these we should warn users that interest calculation can be inaccurate
+YEAR_WARNING = 1
+RATE_WARNING = 50
+ALLOWED_PERCENTAGE = 1
+
+
+@pytest.mark.parametrize(
+    "rate_percent",
+    [1, 10, RATE_WARNING, 100, 327.67, -1, -10, -RATE_WARNING, -100, -327.68],
+)
+@pytest.mark.parametrize("duration_years", [YEAR_WARNING, 2, 3, 100])
+@pytest.mark.parametrize(
+    "balance",
+    [
+        1,
+        1000,
+        10 ** 12,
+        MAX_BALANCE - 1,
+        MAX_BALANCE,
+        MAX_BALANCE + 1,
+        10 ** 21,
+        -1,
+        -1000,
+        -(10 ** 12),
+        MIN_BALANCE + 1,
+        MIN_BALANCE - 1,
+    ],
+)
+def test_interest_calculation_audit(
```

Comment:
This test is hard to read, can we split the case where balance is OOB to avoid all the bools with `MIN_BALANCE <= balance <= MAX_BALANCE`?

Reply:
Splitted